### ISSUE TITLE
[cx] Broken typings

### ISSUE THREAD

**Issue by dakur** [codaxy/cxjs#711]
When I try to build an app using `cx@18.12.7`, I got type errors inside of cx:  
![image](https://user-images.githubusercontent.com/9862581/96873281-086ca880-1475-11eb-907c-2b114395ff0a.png)

I've tried to upgrade to `latest` (v20) and it's even worse:  
![image](https://user-images.githubusercontent.com/9862581/96873014-ab70f280-1474-11eb-9bad-d46ba21fe5b6.png)

I've explored few of them and in v18 there's really no `Layout` file and in v20 `NestedDataView` really does not implement mentioned methods (in typings, in code itself it is).

Can you please fix the typings? Thank you. I would do it, but not pretty sure what to do about all of the errors.

**Comment by mstijak** [comment]
Sure, we'll take a look at it. Can you please tell which tools you're
using? It might be better to avoid passing CxJS code through the TypeScript
compiler, if possible...

**Comment by dakur** [comment]
Thanks for reply.

Basically, I use webpack with babel. Here is my `package.json`:

```json
{
  "dependencies": {
    "babel-polyfill": "^6.23.0",
    "cx": "^18.6.15",
    "cx-react": "^17.10.0",
    "font-awesome": "^4.7.0",
    "moment": "^2.22.2",
    "react": "^16.0.0",
    "react-dom": "^16.0.0",
    "whatwg-fetch": "^2.0.3"
  },
  "devDependencies": {
    "@types/react": "^16.0.28",
    "@types/react-dom": "^16.0.3",
    "@types/uuid": "^3.4.3",
    "@types/webpack-env": "^1.13.3",
    "awesome-typescript-loader": "^3.4.1",
    "babel-core": "^6.25.0",
    "babel-loader": "^7.1.1",
    "babel-plugin-syntax-jsx": "^6.8.0",
    "babel-preset-cx-env": "^17.12.1",
    "babel-preset-env": "^1.6.0",
    "babel-preset-minify": "^0.2.0",
    "copy-webpack-plugin": "^4.0.1",
    "cross-env": "^6.0.0",
    "css-loader": "^0.28.4",
    "extract-text-webpack-plugin": "^3.0.1",
    "file-loader": "^1.1.5",
    "html-webpack-plugin": "^2.29.0",
    "js-joda": "^1.6.2",
    "json-loader": "^0.5.4",
    "neon-js": "^1.1.2",
    "neon-loader": "^1.0.0",
    "node-sass": "^4.5.3",
    "raven-js": "^3.20.1",
    "sass-loader": "^6.0.6",
    "source-map-loader": "^0.2.3",
    "style-loader": "^0.19.0",
    "tslint": "^5.8.0",
    "typescript": "^2.6.2",
    "uuid": "^3.1.0",
    "webpack": "^3.10.0",
    "webpack-dev-server": "^2.9.7",
    "webpack-merge": "^4.1.0"
  }
}
```

How could I avoid using typescript compiler having typescript-based project?

Feel free to ask further questions on my our build-stack if needed.

**Comment by dakur** [comment]
(Note that it's quite old code-base (2 years) which I'm trying to run on current (LTS) node version (12.18.2). Originaly it ran on v9 without issues.)

**Comment by mstijak** [comment]
You might try using webpack's `exclude` and use the compiled version of cx.
Alternatively, it's possible to use babel to compile typescript by simply
stripping type information (babel-preset-typescript). We mostly use that on
TypeScript projects.

**Comment by mstijak** [comment]
I made some changes based on your inputs and published a new beta release. Can you please try it and report your findings?

**Comment by dakur** [comment]
Yeah, that's much better, now it fails only on [this line](https://github.com/codaxy/cxjs/blob/master/packages/cx/src/data/ops/updateTree.d.ts#L3) with:

![image](https://user-images.githubusercontent.com/9862581/97705385-4b053500-1ab4-11eb-99aa-227e0526495a.png)

Thanks.

**Comment by dakur** [comment]
I've made PR #716

**Comment by dakur** [comment]
Yeah, the problem is now resolved with new version 20.10.5. Thanks! 👍
